packages/react/modules/components/configure-adapter/configure-adapter.spec.js

```diff
@@ -328,10 +360,269 @@ describe('lifecycle', () => {
+describe('interacting', () => {
+  let props;
+  let wrapper;
+  const nextAdapterArgs = {
+    user: 'next-user',
+  };
+
+  describe('setAdapterArgs', () => {
+    describe('when configured', () => {
+      beforeEach(() => {
+        props = createTestProps();
+        wrapper = shallow(
+          <ConfigureAdapter {...props}>
+            <ChildComponent />
+          </ConfigureAdapter>
+        );
+
+        wrapper.instance().setAdapterState(AdapterStates.CONFIGURED);
+        wrapper.instance().setAdapterArgs(nextAdapterArgs, { exact: false });
+      });
+
+      it('should update the `state` of `adapterArgs`', () => {
+        expect(wrapper).toHaveState('adapterArgs', nextAdapterArgs);
+      });
+    });
+  });
+
+  describe('setPendingAdapterArgs', () => {
+    beforeEach(() => {
+      props = createTestProps();
+      wrapper = shallow(
+        <ConfigureAdapter {...props}>
+          <ChildComponent />
+        </ConfigureAdapter>
+      );
+    });
+
+    describe('without `pendingAdapterArgs`', () => {
+      beforeEach(() => {
+        wrapper.instance().setPendingAdapterArgs({
+          adapterArgs: nextAdapterArgs,
+          options: { exact: false },
+        });
+      });
+
+      it('should set `pendingAdapterArgs` to `nextAdapterArgs', () => {
+        expect(wrapper.instance().pendingAdapterArgs).toEqual(
+          expect.objectContaining(nextAdapterArgs)
+        );
+      });
+    });
+
+    describe('with `pendingAdapterArgs`', () => {
+      const nextNextAdapterArgs = {
+        firstName: 'user-first-name',
+      };
+      beforeEach(() => {
+        wrapper.instance().setPendingAdapterArgs({
+          adapterArgs: nextAdapterArgs,
+          options: { exact: false },
+        });
+        wrapper.instance().setPendingAdapterArgs({
+          adapterArgs: nextNextAdapterArgs,
+          options: { exact: false },
+        });
+      });
+
+      it('should set `pendingAdapterArgs` merged with `pendingAdapterArgs', () => {
+        expect(wrapper.instance().pendingAdapterArgs).toEqual(
+          expect.objectContaining(nextAdapterArgs)
+        );
+
+        expect(wrapper.instance().pendingAdapterArgs).toEqual(
+          expect.objectContaining(nextNextAdapterArgs)
+        );
+      });
+    });
+  });
+
+  describe('unsetPendingAdapterArgs', () => {
+    describe('with `pendingAdapterArgs`', () => {
+      beforeEach(() => {
+        props = createTestProps();
+        wrapper = shallow(
+          <ConfigureAdapter {...props}>
+            <ChildComponent />
+          </ConfigureAdapter>
+        );
+
+        jest.spyOn(wrapper.instance(), 'setAdapterArgs');
+
+        wrapper.instance().setPendingAdapterArgs({
+          adapterArgs: nextAdapterArgs,
+          options: { exact: false },
+        });
+
+        wrapper.instance().unsetPendingAdapterArgs();
+      });
+
+      it('should invoke `setAdapterArgs`', () => {
+        expect(wrapper.instance().setAdapterArgs).toHaveBeenCalled();
```

Comment:
Nitpick: I usually avoid this test if I also have `toHaveBeenCalledWith`, because the latter implies the function "to have been called".